### ISSUE TITLE
XmlAttribute not applied

### ISSUE THREAD

**Issue by janfockaert** [springfox/springfox#1903]
Latest Swagger2 version 2.7.0

We have a Spring Boot application that exposes XML api and we want to show a correct XML example in the Swagger spec. The @XmlAttribute it is not applied on the spec but shown as an XML element. I already searched for other issues and added the jackson jaxb module.

On the following [sample repo](https://github.com/merapar/spring-boot-swagger-sample) you can reproduce this behaviour.

The expected outcome in Swagger Example model
```xml
<Sample code="1" value="string">
</Sample>
```
but is:
```xml
<Sample>
	<code>1</code>
	<value>string</value>
</Sample>
```

Some code samples copied from the sample repo
Application.java:
```java
@SpringBootApplication
@EnableSwagger2
public class Application implements ApplicationListener<ObjectMapperConfigured> {
	public static void main(String[] args) throws Exception {
		SpringApplication.run(Application.class, args);
	}

	@Bean
	public Docket api() {
		return new Docket(DocumentationType.SWAGGER_2)
				.select()
				.apis(RequestHandlerSelectors.basePackage("com.merapar.controller"))
				.paths(PathSelectors.ant("/*"))
				.build();
	}

	@Override
	public void onApplicationEvent(ObjectMapperConfigured event) {
		event.getObjectMapper().registerModule(new JaxbAnnotationModule());
	}
}
```
Sample.java:
```java
@NoArgsConstructor
@AllArgsConstructor
@XmlRootElement(name = "Sample")
public class Sample {

    @Getter
    @XmlAttribute(name = "value", required = true)
    private String value;

    @Getter
    @XmlAttribute(name = "code")
    private CodeEnum code;

    @XmlEnum
    public enum CodeEnum {
        @XmlEnumValue("1")
        ENUM1(1),
        @XmlEnumValue("2")
        ENUM2(2);

        private int value;

        CodeEnum(int value) {
            this.value = value;
        }

        @Override
        public String toString() {
            return String.valueOf(value);
        }
    }
}
```

Not sure whether this is a bug or a feature request.

Thanks in advance.

**Comment by dilipkrish** [comment]
I suspect it isn't respecting the model in swagger-ui correctly. Not sure what to do about that.

**Comment by janfockaert** [comment]
It's missing the following attributes on the properties with @XmlAttribute:
```json
  "xml": {
    "attribute": true
  }
``` 

so the complete Swagger2 should be:
```json
{
    "swagger": "2.0",
    "info": {
        "description": "Api Documentation",
        "version": "1.0",
        "title": "Api Documentation",
        "termsOfService": "urn:tos",
        "contact": {},
        "license": {
            "name": "Apache 2.0",
            "url": "http://www.apache.org/licenses/LICENSE-2.0"
        }
    },
    "host": "localhost:8080",
    "basePath": "/",
    "tags": [
        {
            "name": "sample-controller",
            "description": "Sample Controller"
        }
    ],
    "paths": {
        "/sample": {
            "get": {
                "tags": [
                    "sample-controller"
                ],
                "summary": "sample",
                "operationId": "sampleUsingGET",
                "consumes": [
                    "application/json"
                ],
                "produces": [
                    "application/xml"
                ],
                "responses": {
                    "200": {
                        "description": "OK",
                        "schema": {
                            "$ref": "#/definitions/Sample"
                        }
                    },
                    "401": {
                        "description": "Unauthorized"
                    },
                    "403": {
                        "description": "Forbidden"
                    },
                    "404": {
                        "description": "Not Found"
                    }
                }
            }
        }
    },
    "definitions": {
        "Sample": {
            "type": "object",
            "required": [
                "value"
            ],
            "properties": {
                "code": {
                    "type": "string",
                    "xml": {
                        "attribute": true
                    },
                    "enum": [
                        "1",
                        "2"
                    ]
                },
                "value": {
                    "type": "string",
                    "xml": {
                        "attribute": true
                    }
                }
            }
        }
    }
}
```

Is this currently not supported than?

Thanks

**Comment by dilipkrish** [comment]
Fixed via #2230
